Sources/Components/Klimabrolet/KlimabroletView.swift

```diff
@@ -0,0 +1,177 @@
+        view.delegate = self
+        return view
+    }()
+
+    private lazy var actionsView: KlimabroletActionsView = {
+        let view = KlimabroletActionsView(withAutoLayout: true)
+        view.layer.shadowOffset = .zero
+        view.delegate = self
+        return view
+    }()
+
+    // MARK: - Initializers
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setup()
+    }
+
+    public override func layoutSubviews() {
+        super.layoutSubviews()
+        closeButton.layer.cornerRadius = closeButton.bounds.width / 2.0
+    }
+
+    // MARK: - Private methods
+
+    private func setup() {
+        backgroundColor = .milk
+        layer.cornerRadius = 20
+        clipsToBounds = true
+
+        scrollView.addSubview(contentView)
+
+        NSLayoutConstraint.activate([
+            contentView.leadingAnchor.constraint(equalTo: scrollView.leadingAnchor),
+            contentView.trailingAnchor.constraint(equalTo: scrollView.trailingAnchor),
+            contentView.widthAnchor.constraint(equalTo: scrollView.widthAnchor),
+            contentView.topAnchor.constraint(equalTo: scrollView.topAnchor),
+            contentView.bottomAnchor.constraint(equalTo: scrollView.bottomAnchor),
+        ])
+
+        addSubview(scrollView)
+        addSubview(actionsView)
+        addSubview(closeButton)
+
+        NSLayoutConstraint.activate([
+            closeButton.topAnchor.constraint(equalTo: topAnchor, constant: .mediumSpacing),
+            closeButton.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -.mediumSpacing),
+
+            scrollView.topAnchor.constraint(equalTo: topAnchor),
+            scrollView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            scrollView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            scrollView.bottomAnchor.constraint(equalTo: actionsView.topAnchor),
+
+            actionsView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            actionsView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            actionsView.bottomAnchor.constraint(equalTo: bottomAnchor)
+        ])
+
+        contentSizeObservation = scrollView.observe(
+            \UIScrollView.contentSize, options: [.old, .new], changeHandler: contentSizeDidChange(_:change:)
+        )
+    }
+
+    @objc private func handleTapOnCloseButton() {
+        delegate?.klimabroletViewDidSelectClose(self)
+    }
+
+    private func contentSizeDidChange(_ scrollView: UIScrollView, change: NSKeyValueObservedChange<CGSize>) {
+        guard let contentSize = change.newValue else {
+            return
+        }
+
+        if scrollView.contentOffset.y + scrollView.frame.height < contentSize.height {
+            actionsView.layer.shadowOpacity = 0.2
+        }
+    }
+}
+
+// MARK: - KlimabroletView UIScrollViewDelegate
```

Comment:
Guess `KlimabroletView` isn't necessary here, since it's the protocol we're interested in. Same for the other ones below.